setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -16,11 +16,25 @@ class ClassesScanner implements ClassesScannerInterface
     protected $excludePatterns = [];
 
     /**
+     * @var array
+     */
+
+    protected $fileResults = [];
+
+    /**
+     * @var DirectoryList
+     */
+
+    protected $directoryList;
+
+    /**
+     * @param DirectoryList $directoryList
      * @param array $excludePatterns
      */
-    public function __construct(array $excludePatterns = [])
+    public function __construct(DirectoryList $directoryList, array $excludePatterns = [])
```

Comment:
Don't get me wrong, continue avoiding it. And please still add any dependencies to the constructor (i.e. DirectoryList), just make them optional as `DirectoryList $directoryList = null` and fall back to the object manager. This way it's fully backwards compatible - and the fallback can be removed later.